mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -2549,7 +2549,16 @@ struct GridwiseGemmAccelRewritePattern
 
     // Obtain data types of inputs.
     auto elementTypeA = op.getA().getType().getElementType();
+    auto maybeElementTypeALoad = getGemmInputElementType(op.getA());
```

Comment:
Would be nice to have a unit-test that uses the `getGemmInputElementType` to test out that logic along with `gridwisegemmtoblockwise`.